pkg/util/iptree/iptree.go

```diff
@@ -0,0 +1,679 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iptree
```

Comment:
Do we really need this?  I get that having thousands of CIDRs might be a problem, but I bet that a simple linear search will be at least as good, maybe better, for 99.999% of users, and when someone has enough CIDRs to need this, we can ask them "why?"

Reply:
talked offline, in a follow up we'll move to a more naive logic and remove the tree, but since is alpha, the code is feature gated and we are in code freeze , we keep it as is